cclient/service.h

```diff
@@ -8,20 +8,27 @@ extern "C" {
 #endif
 
 #include <stdlib.h>
-#include "serializer/serializer.h"
 
 typedef struct {
-  const char* const host;
+  char* contentType;
+  size_t version;
+} iota_headers_t;
```

Comment:
This is client/network protocol specific. No need for this struct? Same goes for `iota_service` etc.
Maybe add a similar vtable for the client interface?